File: remote/src/main/java/com/kelsos/mbrc/viewmodels/MiniControlModel.java
package com.kelsos.mbrc.viewmodels;

import android.graphics.Bitmap;
import com.kelsos.mbrc.annotations.PlayerState;

public class MiniControlModel {
  private Bitmap cover;
  private String title;
  private String artist;
  private String playerState;

  public void setCover(Bitmap cover) {
    this.cover = cover;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public void setArtist(String artist) {
    this.artist = artist;
  }

  public void setPlayerState(@PlayerState.State String playerState) {
    this.playerState = playerState;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/viewmodels/MiniControlModel.kt
package com.kelsos.mbrc.viewmodels

import android.graphics.Bitmap
import com.kelsos.mbrc.annotations.PlayerState
import com.kelsos.mbrc.extensions.empty

class MiniControlModel {
  private var cover: Bitmap? = null
  private var title: String = String.empty
  private var artist: String = String.empty
  @PlayerState.State private var playerState: String = PlayerState.UNDEFINED

  fun setCover(cover: Bitmap?) {
    this.cover = cover
  }

  fun setTitle(title: String) {
    this.title = title
  }

  fun setArtist(artist: String) {
    this.artist = artist
  }

  fun setPlayerState(@PlayerState.State playerState: String) {
    this.playerState = playerState
  }
}
